pkg/odo/util/completion/completionhandlers_test.go

```diff
@@ -18,6 +20,201 @@ import (
 	ktesting "k8s.io/client-go/testing"
 )
 
+func TestServicePlanCompletionHandler(t *testing.T) {
+	serviceClassList := &scv1beta1.ClusterServiceClassList{
+		Items: []scv1beta1.ClusterServiceClass{testingutil.FakeClusterServiceClass("class name", "dummy")},
+	}
+	tests := []struct {
+		name                 string
+		returnedServiceClass *scv1beta1.ClusterServiceClassList
+		returnedServicePlan  []scv1beta1.ClusterServicePlan
+		output               []string
+		parsedArgs           parsedArgs
+	}{
+		{
+			name: "Case 0: no service name supplied",
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create"},
+				},
+			},
+			output: []string{},
+		},
+		{
+			name:                 "Case 1: single plan exists",
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan:  []scv1beta1.ClusterServicePlan{testingutil.FakeClusterServicePlan("default", 1)},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+			},
+			output: []string{"default"},
+		},
+		{
+			name:                 "Case 2: multiple plans exist",
+			returnedServiceClass: serviceClassList,
+			returnedServicePlan: []scv1beta1.ClusterServicePlan{
+				testingutil.FakeClusterServicePlan("plan1", 1),
+				testingutil.FakeClusterServicePlan("plan2", 2),
+			},
+			parsedArgs: parsedArgs{
+				original: complete.Args{
+					Completed: []string{"create", "class name"},
+				},
+			},
+			output: []string{"plan1", "plan2"},
+		},
+	}
+
+	for _, tt := range tests {
+		client, fakeClientSet := occlient.FakeNew()
+		context := genericclioptions.NewFakeContext("project", "app", "component", client)
+
+		fakeClientSet.ServiceCatalogClientSet.PrependReactor("list", "clusterserviceclasses", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, tt.returnedServiceClass, nil
+		})
+
+		fakeClientSet.ServiceCatalogClientSet.PrependReactor("list", "clusterserviceplans", func(action ktesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, &scv1beta1.ClusterServicePlanList{Items: tt.returnedServicePlan}, nil
+		})
+
+		completions := ServicePlanCompletionHandler(nil, tt.parsedArgs, context)
+		sort.Strings(completions)
+
+		// Sort the output and expected o/p in-order to avoid issues due to order as its not important
+		sort.Strings(completions)
```

Comment:
why are we sorting twice? I didn't understand from the comment above

Reply:
Damn it... I wanted to sort the expected here. Good catch!

Reply:
Fixed

Reply:
:+1: